internal/api/service/service.go

```diff
@@ -9,18 +9,28 @@ import (
 	"k8s.io/helm/pkg/proto/hapi/release"
 )
 
-func New(l ReleaseLister) *Service {
+func New(l ReleaseLister, git GitCommands, repo string, ref string) *Service {
 	return &Service{
-		lister: l,
+		lister:     l,
+		client:     git,
+		repository: repo,
+		ref:        ref,
 	}
 }
 
 type ReleaseLister interface {
 	ListAll(namespace string) ([]models.Release, error)
 }
 
+type GitCommands interface {
+	GetTravisCIBuildURLForRef(ctx context.Context, repo string, ref string) (string, error)
+}
+
 type Service struct {
-	lister ReleaseLister
+	lister     ReleaseLister
+	client     GitCommands
```

Comment:
`client` & `GitCommands` leak implementation details of the interface unnecessarily. An interface should provide a black box abstraction. When we use it, we needn't care that it's implemented by a `github.Client` and we don't even need to know it's related to git/github at all.

As far as we (the `Service` consuming the interface) need to know, the interface integrates with the Travis Checks API. So, I'd suggest naming the interface `TravisChecker` (go interfaces are [conventionally](https://golang.org/doc/effective_go.html#interface-names) named this way), and the struct field `travis`.